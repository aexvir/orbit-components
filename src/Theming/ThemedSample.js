// @flow
import React from "react";
import { withTheme } from "theming";
import defaultTheme from "orbit-design-token";

type ComponentProps = {
  theme: typeof defaultTheme,
};

const Component = ({ theme }: ComponentProps) => (
  <div style={{ color: theme.colorTextPrimary, fontFamily: theme.fontFamily }}>
    {theme.colorTextPrimary}
  </div>
);

// We're passing a default theme for Component that aren't wrapped in the ThemeProvider
Component.defaultProps = {
  theme: defaultTheme, // eslint-disable-line react/default-props-match-prop-types
};

// Themed component cannot be used without ThemeProvider
const ThemedComponent = withTheme(Component);
export { ThemedComponent as default, Component as RawComponent };
