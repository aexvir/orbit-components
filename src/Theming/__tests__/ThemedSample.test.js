// @flow

import * as React from "react";
import { shallow } from "enzyme";
import defaultTheme from "orbit-design-token";

import { RawComponent } from "../ThemedSample";

describe("RawComponent", () => {
  const component = shallow(<RawComponent />);
  const div = component.find("div");

  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should contain a default style", () => {
    const expected = `color:${defaultTheme.colorTextPrimary};font-family:${
      defaultTheme.fontFamily
    }`;
    expect(div.render().attr("style")).toBe(expected);
  });

  it("Should contain a color", () => {
    expect(div.render().text()).toBe(defaultTheme.colorTextPrimary);
  });
});
