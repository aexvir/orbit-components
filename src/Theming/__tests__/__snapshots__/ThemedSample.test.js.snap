// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RawComponent should match snapshot 1`] = `
<div
  style={
    Object {
      "color": "#46515e",
      "fontFamily": "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
    }
  }
>
  #46515e
</div>
`;
